src/sw.test.js

```diff
@@ -0,0 +1,198 @@
+/* eslint-disable import/no-unresolved */
+/* eslint-disable import/first */
+import fs from 'fs';
+import { join, resolve } from 'path';
+
+const serviceWorker = fs.readFileSync(join(__dirname, '..', 'public/sw.js'));
+
+const serviceWorkerCode = `${serviceWorker.toString()}
+export { fetchEventHandler };
+`;
+
+fs.writeFileSync(
+  resolve(__dirname, 'service-worker-test.js'),
+  serviceWorkerCode,
+);
+
+describe('Service Worker', () => {
+  const originalFetch = global.fetch;
+  const fetchSpy = jest.spyOn(global, 'fetch');
+  let fetchEventHandler;
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    global.fetch = originalFetch;
+  });
+
+  afterAll(() => {
+    fs.unlinkSync(resolve(__dirname, 'service-worker-test.js'));
```

Comment:
Code Climate does not appear to like this file being deleted (and causes the PR checks to fail... perhaps we should leave it - it is already git ignored